### PR TITLE
Do not change or generate README.md after running rspec

### PR DESCRIPTION
**Pull request by nafu**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Description

Do not change or generate README.md after running rspec.
If anybody knows better way for handling this, please feedback to this PR. 🙏 

### Motivation and Context

After running rspec, it generates some README.md on my machine that is not needed without this patch.

### CODE REVIEW

File: fastlane/spec/cli_tools_distributor_spec.rb [hunk 1]
```diff
@@ -50,6 +50,8 @@
       expect do
         Fastlane::CLIToolsDistributor.take_off
       end.to raise_error(SystemExit)
+
+      `git checkout fastlane/README.md`
```
Comment (KrauseFx) [inline fastlane/spec/cli_tools_distributor_spec.rb]: That seems like a hack, can we just not generate the file in the first place instead?

File: fastlane/spec/commands_generator_spec.rb [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
+
 describe Fastlane::CommandsGenerator do
   before(:each) do
     ENV['DOTENV'] = nil
-    fastlane_folder = File.absolute_path('./fastlane/spec/fixtures/dotenvs/withFastfiles/parentonly/fastlane')
     allow(FastlaneCore::FastlaneFolder).to receive(:path).and_return(fastlane_folder)
   end
 
+  after(:all) do
+    FileUtils.rm(File.join(fastlane_folder, 'README.md'))
```
Comment (KrauseFx) [inline fastlane/spec/commands_generator_spec.rb]: Same here

File: fastlane/spec/lane_manager_spec.rb [hunk 3]
```diff
@@ -70,7 +72,11 @@ def ensure_dot_env_value_from_folders(folders, envs, expected_values)
 
       describe "successfull init" do
         before do
-          allow(FastlaneCore::FastlaneFolder).to receive(:path).and_return(File.absolute_path('./fastlane/spec/fixtures/fastfiles/'))
+          allow(FastlaneCore::FastlaneFolder).to receive(:path).and_return(fastlane_folder)
```
Comment (KrauseFx) [inline fastlane/spec/lane_manager_spec.rb]: Same here

**Review by KrauseFx — CHANGES_REQUESTED** [review]
I noticed those files too, and thought it's related to my machine somehow. Instead of adding those workarounds, can we just not generate those files at all instead? Maybe just mock the method that writes out the file?

File: fastlane/spec/commands_generator_spec.rb [hunk 4]
```diff
@@ -5,6 +5,8 @@
     ENV['DOTENV'] = nil
     fastlane_folder = File.absolute_path('./fastlane/spec/fixtures/dotenvs/withFastfiles/parentonly/fastlane')
     allow(FastlaneCore::FastlaneFolder).to receive(:path).and_return(fastlane_folder)
+    allow(FastlaneCore::Env).to receive(:truthy?).and_return(:default)
+    allow(FastlaneCore::Env).to receive(:truthy?).with('FASTLANE_SKIP_DOCS').and_return(true)
```
Comment (KrauseFx) [inline fastlane/spec/commands_generator_spec.rb]: Should we set this for all tests inside the spec_helper?

File: fastlane/spec/cli_tools_distributor_spec.rb [hunk 5]
```diff
@@ -43,6 +43,8 @@
     end
 
     it "checks for updates even if the lane has an error" do
+      allow(FastlaneCore::Env).to receive(:truthy?).and_return(:default)
+      allow(FastlaneCore::Env).to receive(:truthy?).with('FASTLANE_SKIP_DOCS').and_return(true)
```
Comment (nafu) [inline fastlane/spec/cli_tools_distributor_spec.rb]: There is 3 duplication codes, so it is better to define helper in `fastlane_core/spec/spec_helper.rb` or `fastlane/spec/spec_helper.rb`?

File: fastlane/spec/commands_generator_spec.rb (hunk 4 above)
Comment (nafu) [inline fastlane/spec/commands_generator_spec.rb]: Yes 👍

Comment (nafu) [inline fastlane/spec/commands_generator_spec.rb]: @KrauseFx Uhh, no after all. I tried to define these in spec_helper, and it had other problems to cause test failure.

**Review by giginet — CHANGES_REQUESTED** [review]
I think this solution is strange.

You should not stub this environment variables.

It's natural to skip executing DocsGenerator.run in testing.

https://github.com/fastlane/fastlane/blob/60bac7e4a2a6c1e19d1cd20eb83201e7d0cb9e48/fastlane/lib/fastlane/lane_manager.rb#L64-L64

```
Fastlane::DocsGenerator.run(ff) if !Helper.test? && !FastlaneCore::Env.truthy?("FASTLANE_SKIP_DOCS")
```

**Comment by nafu** [comment]
Hmm, CircleCI select Xcode 7.0 😓 

image


https://circleci.com/gh/fastlane/fastlane/8447

**Comment by nafu** [comment]
@KrauseFx @giginet 
I addressed the feedbacks, could you review again?

**Review by KrauseFx — APPROVED** [review]
Looks great :+1: Thanks @nafu

**Review by giginet — APPROVED** [review]

**Comment by giginet** [comment]
@nafu Sorry for my late reply.
You can merge it 😄

**Comment by nafu** [comment]
@KrauseFx @giginet Thanks 🚀

**Pull request merged**